src/Core/Cache/MemcachedCacheDriver.php

```diff
@@ -62,13 +59,14 @@ public function __construct(array $memcached_hosts)
 	 */
 	public function getAllKeys($prefix = null)
 	{
-		// Doesn't work because of https://github.com/php-memcached-dev/php-memcached/issues/367
-		// returns everytime an empty array
-		throw new InternalServerErrorException('getAllKeys for Memcached not supported yet');
+		$keys = $this->getOriginalKeys($this->memcached->getAllKeys());
 
-		$list = $this->getOriginalKeys($this->memcached->getAllKeys());
-
-		return $this->filterPrefix($list, $prefix);
+		if ($this->memcached->getResultCode() == Memcached::RES_SUCCESS) {
+			return $this->filterArrayKeysByPrefix($keys, $prefix);
+		} else {
+			logger('Memcached \'getAllKeys\' failed with ' . $this->memcached->getResultMessage(), LOGGER_ALL);
+			return null;
```

Comment:
Isn't this bound to trigger the same kind of error that prompted the fix in the first place?